libs/js-sdk/src/effective-feature-state.ts

```diff
@@ -10,14 +10,33 @@ export interface EffectiveFeatureValues {
 type FeatureValue = string | boolean | number | undefined
 
 export class EffectiveFeaturesState {
+    private static instance: EffectiveFeaturesState | null = null
     private valueUpdatedCallbacks: Array<
         (featureKey: string, value: FeatureValue) => void
     > = []
 
-    constructor(
+    private constructor(
         public audiences: string[],
         public store: EffectiveFeatureStore = new MemoryEffectiveFeatureStore(),
-    ) {}
+    ) {
+        if (EffectiveFeaturesState.instance) {
+            return EffectiveFeaturesState.instance
+        }
+        EffectiveFeaturesState.instance = this
+    }
+
+    static getInstance(
+        audiences: string[],
```

Comment:
I think this might cause an issue when the audiences change because they won't update.

Possibly we need to hoist `valueUpdatedCallbacks` and pass it into the constructor?